File: ios/Data/FirebaseService.swift
import Foundation
import FirebaseFirestore

class FirebaseService {
    private let db = Firestore.firestore()

    func saveGroup(_ group: Group) {
        let data: [String: Any] = [
            "id": group.id,
            "name": group.name,
            "members": group.members
        ]
        db.collection("groups").document(group.id).setData(data)
    }

    func getGroups(completion: @escaping ([Group]) -> Void) {
        db.collection("groups").getDocuments { (snapshot, error) in
            guard let documents = snapshot?.documents else {
                completion([])
                return
            }
            let groups = documents.compactMap { doc -> Group? in
                let data = doc.data()
                guard let id = data["id"] as? String,
                      let name = data["name"] as? String,
                      let members = data["members"] as? [String] else { return nil }
                return Group(id: id, name: name, members: members)
            }
            completion(groups)
        }
    }
}

File: android/app/src/main/java/com/lumoralabs/macro/data/FirebaseService.kt
package com.lumoralabs.macro.data

import com.google.firebase.firestore.FirebaseFirestore

object FirebaseService {
    private val db = FirebaseFirestore.getInstance()

    fun saveGroup(group: Group) {
        db.collection("groups").document(group.id).set(group)
    }

    fun getGroups(onResult: (List<Group>) -> Unit) {
        db.collection("groups").get().addOnSuccessListener { result ->
            val groups = result.documents.mapNotNull { it.toObject(Group::class.java) }
            onResult(groups)
        }
    }
}
